checker/raw_result.go

```diff
@@ -243,3 +244,46 @@ type WorkflowJob struct {
 	Name *string
 	ID   *string
 }
+
+// TokenPermissionsData represents data about a permission failure.
+type TokenPermissionsData struct {
+	TokenPermissions []TokenPermission
+}
+
+// PermissionLocation represents a declaration type.
+type PermissionLocation string
+
+const (
+	// PermissionLocationTop is top-level workflow permission.
+	PermissionLocationTop PermissionLocation = "topLevel"
+	// PermissionLocationJob is job-level workflow permission.
+	PermissionLocationJob PermissionLocation = "jobLevel"
+)
+
+// PermissionType represents a permission type.
+type PermissionType string
+
+const (
+	// PermissionTypeUndeclared is an undecleared permission.
+	PermissionTypeUndeclared PermissionType = "undeclared"
+	// PermissionTypeWrite is a permission set to `write` for a permission we consider potentially dangerous.
+	PermissionTypeWrite PermissionType = "write"
+	// PermissionTypeRead is a permission set to `read`.
+	PermissionTypeRead PermissionType = "read"
+	// PermissionTypeNone is a permission set to `none`.
+	PermissionTypeNone PermissionType = "none"
+	// PermissionTypeUnknown is for other kinds of alerts, mostly to support debug messages.
+	// TODO: remove it once we have implemented severity (#1874).
+	PermissionTypeUnknown PermissionType = "unknown"
+)
```

Comment:
Shouldn't this be `PermissionLevel` and `PermissionType` should instead be - "content", "status" etc. ?

Reply:
thanks, will rename!